src/dask_awkward/lib/io/io.py

```diff
@@ -674,3 +675,49 @@ def bytes_reading_ingredients(
         sample_bytes = sample_bytes[:rfind]
 
     return out, sample_bytes
+
+
+class FromTextFn:
+    def __init__(self):
+        pass
+
+    def __call__(self, ingredients: tuple) -> ak.Array:
+        (fs, path, compression, offsets, length, delimiter) = ingredients
+
+        with fs.open(path, compression=compression) as f:
+            if offsets == 0 and length is None:
+                bytestring = f.read()
+            else:
+                bytestring = read_block(f, offsets, length, delimiter)
+
+            buffer = np.frombuffer(bytestring, dtype=np.uint8)
+            array = ak.from_numpy(buffer)
+            array = ak.unflatten(array, len(array))
+            array = ak.enforce_type(array, "string")
+            array_split = ak.str.split_pattern(array, "\n")
+            lines = array_split[0]
```

Comment:
-> list[string]? but what was array_split, then, why the [0]?

Reply:
`array_split` ends up being of type `1 * var * string`; so by grabbing `[0]` we get an array of strings `N * string` (used the awkward docs https://awkward-array.org/doc/main/user-guide/how-to-strings-read-binary.html)